netlink-packet-route/src/rtnl/link/nlas/link_infos.rs

```diff
@@ -478,6 +500,356 @@ impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoKind {
     }
 }
 
+struct IpAddrNla {
+    addr: IpAddr,
+    index: u16,
+}
```

Comment:
This is specific to bond. Please rename it to `BondArpIPTargetNla`. And also a comment like this could help code next  reviewer/maintainer or yourself in 10+ months:

The `bond_fill_info()` in kernel is using index of IP address array as NLA kind, but it was ignored in `bond_changelink()`. So we only exposing the `IpAddr` and use this internal struct for serialization and deserialization.